# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_grub.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- grub-0.93/lib/device.c.orig	Thu Sep  4 16:25:24 2003
+++ grub-0.93/lib/device.c	Thu Sep  4 16:23:46 2003
@@ -699,7 +699,7 @@
       return 0;
     }
   
-#if defined(__linux__) && (!defined(__GLIBC__) || \
+#if defined(__linux__) && (!defined(__dietlibc__)) && (!defined(__GLIBC__) || \
         ((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1))))
   /* Maybe libc doesn't have large file support.  */
   {
--- grub-0.93/grub/asmstub.c~	Mon Dec  2 23:20:45 2002
+++ grub-0.93/grub/asmstub.c	Thu Sep  4 16:30:09 2003
@@ -934,7 +934,7 @@
     return BIOSDISK_ERROR_GEOMETRY;
 
   /* Seek to the specified location. */
-#if defined(__linux__) && (!defined(__GLIBC__) || \
+#if defined(__linux__) && (!defined(__dietlibc__)) && (!defined(__GLIBC__) || \
 	((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1))))
   /* Maybe libc doesn't have large file support.  */
   {
